plugins/outputs/inlong/inlong.go

```diff
@@ -0,0 +1,88 @@
+package inlong
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+
+	"github.com/apache/inlong/inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/dataproxy"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const managerURLSuffix = "/inlong/manager/openapi/dataproxy/getIpList"
+
+type Inlong struct {
+	GroupID    string          `toml:"group_id"`
+	StreamID   string          `toml:"stream_id"`
+	ManagerURL string          `toml:"manager_url"`
+	Log        telegraf.Logger `toml:"-"`
+
+	producerFunc func(groupId string, managerUrl string) (dataproxy.Client, error)
+	producer     dataproxy.Client
+	serializer   telegraf.Serializer
+}
+
+func (*Inlong) SampleConfig() string {
+	return sampleConfig
+}
+
+func (i *Inlong) SetSerializer(serializer telegraf.Serializer) {
+	i.serializer = serializer
+}
+
+func (i *Inlong) Connect() error {
+	producer, err := i.producerFunc(i.GroupID, i.ManagerURL+managerURLSuffix)
+	if err != nil {
+		return &internal.StartupError{Err: err, Retry: true}
+	}
+	i.producer = producer
+	return nil
+}
+
+func (i *Inlong) Close() error {
+	i.producer.Close()
+	return nil
+}
+
+func (i *Inlong) Write(metrics []telegraf.Metric) error {
```

Comment:
What happens if the Inlong instance goes down for whatever reason and comes up again e.g. due to network issues or service outages? Would the plugin reconnect automatically in this case?

Reply:
Yes, SDK has the ability to automatically reconnect